internal/api/v1/applications.go

```diff
@@ -344,12 +373,106 @@ func (hc ApplicationsController) Update(w http.ResponseWriter, r *http.Request)
 		return NewBadRequest("instances param should be integer equal or greater than zero")
 	}
 
-	workload := application.NewWorkload(cluster, app.AppRef())
-	err = workload.Scale(r.Context(), updateRequest.Instances)
+	app, err := application.Lookup(ctx, cluster, org, appName)
 	if err != nil {
 		return InternalError(err)
 	}
 
+	// TODO: Can we optimize to perform a single restart regardless of what changed ?!
+	// TODO: Should we ?
```

Comment:
Interesting, I think we should but it can be tricky in some cases. I'm not sure if all the information is kept on one object (the `Deployment`). We can keep mutating the receiver of those methods (the "app") until we are ready and then apply the thing once, but if there are more than one kubernetes objects involved, the applying method can't fail midway and leave things in an inconsistent state.

Reply:
Yep. Hence the more questing tone of the comment. Not sure if possible, and if yes, if it would be worth it. The currently 3 changes (at most) might even cause at kube level only a single restart anyway, if the deployment controller sees them quickly enough.